R/connect.R

```diff
@@ -99,8 +103,19 @@ setMethod("dbConnect", "MariaDBDriver",
       bigint = bigint
     )
 
-    dbExecute(conn, "SET time_zone = '+00:00'")
-    dbExecute(conn, "SET autocommit = 0")
+    if (!missing("timezone")) {
+      tryCatch(
+        dbExecute(conn, paste0("SET time_zone = '", timezone, "'"))
+        ,
+        error = function(e) {
+          dbExecute(conn, "SET time_zone = '+00:00'")
+          message("Unknown or incorrect time zone: '", timezone, "'")
+        }
+      )
+    } else {
+      dbExecute(conn, "SET time_zone = '+00:00'")
+    }
+
```

Comment:
I misread -- failure to set the timezone really should be an error.

```suggestion
    dbExecute(conn, paste0("SET time_zone = ", dbQuoteString(conn, timezone))
    dbExecute(conn, "SET autocommit = 0")
```